File: lab8/fromClassRepository/lab8demo.py
# Bob Roos
# Lab 8 demo--object-oriented programming in Python
# This program requires files pirate.py and parrot.py
#
# To run this program, type:
#     python lab8demo.py

from pirate import *
from parrot import *

"""
  main program creates a Pirate and his Parrot and makes them speak
"""
if __name__ == "__main__":
   name = raw_input("Pirate's name? ")
   pirate = Pirate(name)

   name = raw_input(pirate.getName()+"'s parrot's name? ")
   parrot = Parrot(name)

   print (pirate.getName()+" says "+pirate.speak())
   print (pirate.getName()+"'s parrot, "+parrot.getName()+
           ", says "+parrot.speak())

File: lab8/fromClassRepository/lab8demo.cpp
// Bob Roos
// Demonstration program for lab 8. This program is used in
// conjunction with the following files:
//       pirate.h
//       pirate.cpp
//       parrot.h
//       parrot.cpp
//
// For a "single file" version of this example, see file "lab8onefile.cpp".
//
// To compile and execute it, type:
//       g++ lab8demo.cpp pirate.cpp parrot.cpp -o lab8demo
//       ./lab8demo

#include <iostream>
#include <string>
#include "pirate.h"
#include "parrot.h"

using namespace std;

int main() {
   string name;

   cout << "Pirate's name? ";
   cin >> name;
   Pirate pirate(name);

   cout << pirate.getName() << "'s parrot's name? ";
   cin >> name;
   Parrot parrot(name);

   cout << pirate.getName() << " says " << pirate.speak() << endl;
   cout << pirate.getName() << "'s parrot, " << parrot.getName() << ", says " 
        << parrot.speak() << endl;
}
